core/matrix/identity.cpp

```diff
@@ -54,12 +54,21 @@ template <typename ValueType>
 void Identity<ValueType>::apply_impl(const LinOp* alpha, const LinOp* b,
                                      const LinOp* beta, LinOp* x) const
 {
-    precision_dispatch_real_complex<ValueType>(
-        [this](auto dense_alpha, auto dense_b, auto dense_beta, auto dense_x) {
-            dense_x->scale(dense_beta);
-            dense_x->add_scaled(dense_alpha, dense_b);
-        },
-        alpha, b, beta, x);
+    if (auto bI = dynamic_cast<const Identity<ValueType>*>(b)) {
+        if (auto xs = dynamic_cast<EnableScaledIdentityAddition*>(x)) {
+            xs->add_scaled_identity(alpha, beta);
+        } else {
+            GKO_NOT_IMPLEMENTED;
+        }
+    } else {
+        precision_dispatch_real_complex<ValueType>(
+            [this](auto dense_alpha, auto dense_b, auto dense_beta,
+                   auto dense_x) {
+                dense_x->scale(dense_beta);
+                dense_x->add_scaled(dense_alpha, dense_b);
+            },
+            alpha, b, beta, x);
+    }
```

Comment:
From our internal poll, it seems like this interface option was not really popular. Can we remove it for now and maybe come back to it later?

Reply:
Out of the 3 votes, two were for including this. I think you wanted purely `add_scaled_identity` only, while @yhmtsai and @tcojean thought having both would be good.